manifests/resource/mailhost.pp

```diff
@@ -77,13 +75,13 @@
 #  }
 #
 define nginx::resource::mailhost (
-  Integer $listen_port,
+  Stdlib::Port $listen_port,
   Enum['absent', 'present'] $ensure              = 'present',
   Variant[Array[String], String] $listen_ip      = '*',
   Optional[String] $listen_options               = undef,
   Boolean $ipv6_enable                           = false,
   Variant[Array[String], String] $ipv6_listen_ip = '::',
```

Comment:
I don't know if this should also be tightened. Not sure about the nginx spec for listening.